### PR TITLE
Enable jenkins enviroment to run failed test cases

### PR DESCRIPTION
**Pull request by bilalawan321**

### CODE REVIEW

**Comment by bilalawan321** [comment]
@nedo i have tested on AWS UI and its working fine and now i have to test it on master, 
Can you please Review this PR. ?

**Comment by nedo** [comment]
@bilalawan321 please get this pass through pep8 and pylint first. `continuous-integration/travis-ci/pr` job should also Pass.

**Comment by bilalawan321** [comment]
@nedo all pep8 and pylint issues have been fixed, and **Travis** job is also passing successfully.

**Comment by bilalawan321** [comment]
@farhan-arshad-dev can you review this PR ?

File: tests/common/globals.py [hunk 1]
```diff
@@ -70,24 +70,14 @@ def __init__(self, project_log):
...
         self.fifteenth_existence = 14
         self.sixteenth_existence = 15
+        self.jenkins = True
 
-        # Read  user_preferences.yml and set globals accordingly
-        with open("user_preferences.yml") as user_file:
-            user_preferences = yaml.safe_load(user_file)
-            user_file.close()
+        # Read user_preferences.yml and set globals accordingly
+        self.set_target_environments()
```
Comment (farhan-arshad-dev) [inline tests/common/globals.py:76]: IMO: `setup_global_environment` is more appropriate.

File: tests/common/globals.py [hunk 2]
```diff
@@ -100,6 +90,29 @@ def __init__(self, project_log):
...
+            self.target_environment = user_preferences.get('Settings').get('target_environment')
+            self.server_url = user_preferences.get('Settings').get('appium_server')
+            self.android_platform_version = user_preferences.get('Settings').get('android_platform_version')
+            self.ios_platform_version = user_preferences.get('Settings').get('ios_platform_version')
+            self.aut_current_path = user_preferences.get('Settings').get('aut_current_path')
+            self.aut_latest_path = user_preferences.get('Settings').get('aut_latest_path')
+            self.login_user_name = user_preferences.get('User').get('login_user_name')
+            self.login_password = user_preferences.get('User').get('login_password')
+
+        elif self.jenkins is True:
```
Comment (farhan-arshad-dev) [inline tests/common/globals.py:112]: no need to check `self.jenkins is True`.   use else statement

File: tests/conftest.py [hunk 3]
```diff
@@ -187,25 +193,25 @@ def login(set_capabilities, setup_logging):
...
-            'onclick="window.open(this.src)" align="right"/></div>'.format(file_path)
-        extra.append(pytest_html.extras.html(html))
+    # if report.failed:
+    # test_case_name = str(item)
+    # file_name = test_case_name[test_case_name.find(' '):-1] + '.png'
+    # file_path = SharedData.screenshots_directory + '/' + file_name
+    # SharedData.driver.save_screenshot(file_path)
+    # html = '<div><img src="{}" alt="screenshot" style="width:304px;height:228px;" ' \
+    #     'onclick="window.open(this.src)" align="right"/></div>'.format(file_path)
+    # extra.append(pytest_html.extras.html(html))
```
Comment (farhan-arshad-dev) [inline tests/conftest.py:214]: if these comments are unnecessary please remove them.

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 3 above)
Comment (bilalawan321) [reply tests/conftest.py:214]: @farhan-arshad-dev  this test case is failing on jenkins master job, 
i will fix this test case when i will fix all other failed test cases, in a saperate PR.

**Review by farhan-arshad-dev — APPROVED** [review]

**Pull request merged**